src/scipp/core/bins.py

```diff
@@ -133,6 +133,33 @@ def __itruediv__(self, lut: lookup):
         _cpp.buckets.scale(self._obj, _cpp.reciprocal(lut.func), lut.dim)
         return self
 
+    def __getitem__(self, key):
+        dim, index = key
+        if isinstance(index, _cpp.Variable):
+            if index.ndim == 0:
+                return self._obj.group(concat([index], dim)).squeeze(dim)
```

Comment:
Hmm, I thought I should be able to use `flatten`, but for 0-D it falls back to `broadcast`, which means it will also think it is non-contiguous (i.e., has a stride of 0). Maybe we should fix `flatten` to return something with stride 1 (of length 1) instead?

Reply:
Done, please have another look @jl-wynen

Reply:
This now means that the variable can have any dimensions as long as it has only a single element. This is inconsistent with value based slicing on the data array itself and I don't think we should allow it.

Reply:
Which variable? The key is checked for ndim==0.

Reply:
Missed the `if`, sorry